Cargo.toml

```diff
@@ -32,6 +32,7 @@ heapless       = { version = "0.8", optional = true }
 libc           = { version = "0.2.153", default-features = false, optional = true } # 0.2.79 is the first version that has IP_PMTUDISC_OMIT
 parking_lot    = { version = "0.12", optional = true }
 moka           = { version = "0.12.3", optional = true, features = ["future"] }
+openssl        = { version = "0.10.57", optional = true } # 0.10.57 upgrades to 'bitflags' 2.x
```

Comment:
Well, ring and openssl offer equivalent functionality; users of `domain` can choose which one they prefer.  So we will have to add feature gates for enabling one or the other.

Reply:
At present they don't offer equivalent functionality though, you need openssl to do signing because `ECDSAP256SHA256` is `MUST` for signing according to https://datatracker.ietf.org/doc/html/rfc8624#section-3.1 but we can currently only use externally created ECDSAP256SHA256 keys with OpenSSL, not with Ring.

Reply:
That's true: Ring's interface makes it impossible to import ECDSA keys from the BIND private key format.  However, users who would like to avoid OpenSSL can still opt to disable it and only use other kinds of keys.  As long as Ring's limitations are adequately described in the documentation, I think we should offer both.

Reply:
They can't avoid OpenSSL and do compliant signing or validation as that requires `ECDSAP256SHA256`. Perhaps indeed there are use cases which wouldn't need full compliance, but we should be careful not to make a footgun for users of the library. IMO.

Reply:
Actually, I've rewritten the interface so that ECDSA works on Ring.  Unlike `ldns`, our API now requires key signing to load _both_ the public and private keys from files at startup.  I was trying to avoid this since it seemed too brittle to require both files to be present (especially since most people would assume you only need the private key file), but it has the added benefit of letting us check for corruption or mismatches.

Reply:
I've now introduced separate features for OpenSSL and Ring.  Users can choose to disable both (... once `unstable-validate` doesn't unconditionally require Ring) and bring their own cryptography.